docs/tealscript-migration.md

```diff
@@ -0,0 +1,587 @@
+  applicationID: app,
+  methodArgs: ['algo dev'],
+});
+
+assert(result === 'hello algo dev')
+```
+
+##### Algorand TypeScript
+
+```ts
+const result = abiCall(Hello.prototype.greet, {
+  appId: app,
+  args: ['algo dev'],
+}).returnValue
+assert(result === 'hello algo dev')
+```
+
+#### App Creation
+
+In Algorand TypeScript, you must first explicitly compile a contract before creating it or access the programs/schema
+
+##### TEALScript
+
+```ts
+sendMethodCall<typeof Greeter.prototype.createApplication>({
+  clearStateProgram: Greeter.clearProgram(),
+  approvalProgram: Greeter.approvalProgram(),
+  globalNumUint: Greeter.schema.global.numUint,
+  methodArgs: ['hello'],
+});
+
+const app = this.itxn.createdApplicationId;
+
+const result = sendMethodCall<typeof Greeter.prototype.greet>({
+  applicationID: app,
+  methodArgs: ['world'],
+});
+
+assert(result == 'hello world')
+```
+
+##### Algorand TypeScript
+
+```ts
+// First explicitly compile the app
+const compiled = compileArc4(Greeter)
+
+const app = compiled.call.createApplication({
+  args: ['hello'],
+  globalNumUint: compiled.globalUints
+}).itxn.createdApp
+
+const result = compiled.call.greet({
+  args: ['world'],
+  appId: app,
+}).returnValue
+
+assert(result === 'hello world')
+```
+
+### Reference Types
+
+TODO: For 1.0 we will probably have similar types to TEALScript (i.e. `AppID` that is an ABI `uint64`) but exact API is TBD
+
+### Compiled Contract Information
+
+TEALScript contracts have static methods for getting the contract programs and schema. In Algorand TypeScript, you must first explicitly
+compile the contract and then use the resulting object to access program information.
+
+##### TEALScript
+
+```ts
+// Access program information directly via static methods
+sendMethodCall<typeof Greeter.prototype.createApplication>({
+  clearStateProgram: Greeter.clearProgram(),
+  approvalProgram: Greeter.approvalProgram(),
+  globalNumUint: Greeter.schema.global.numUint,
+  methodArgs: ['hello'],
+});
+```
+
+##### Algorand TypeScript
+
+```ts
+// First explicitly compile the app
+const compiled = compileArc4(Greeter)
+
+// Then access program information on the compiled object
+const app = compiled.call.createApplication({
+  args: ['hello'],
+  globalNumUint: compiled.globalUints
+}).itxn.createdApp
+```
+
+### Logic Sigs
+
+In TEALScript, logic sigs must implement the `logic` method which may take one or more arguments which map to the lsig arguments when
+forming the transaction. All lsigs are approved unless an error occurs. Algorand TypeScript also requires implementation of the `program` method but it may not take an arguments and must return a `boolean` or `uint64` indicating whether the transaction is approved or not.
+
+##### TEALScript
+
+```ts
+class DangerousPaymentLsig extends LogicSig {
+  logic(amt: uint64) {
+    assert(this.txn.amount === amt)
+  }
+}
+```
+
+##### Algorand TypeScript
+
+TODO: How do you get args?
```

Comment:
we can use `op.arg(n)` to get the arguments.
e.g. https://github.com/algorandfoundation/algorand-typescript-testing/blob/main/examples/htlc-logicsig/signature.algo.ts#L27